### DISCUSSION TITLE
Milvus 单服务器部署

### DISCUSSION THREAD

**Discussion by zs001122** [milvus-io/milvus#37474] [Q&A and General discussion]
2.0以后的版本在检索上的性能差异大吗？rel7的机器在做源码编译的时候（2.4.15）太多问题。我看分支能用的rpm包只有2.0.2

**Comment by yhmo** [comment]
rpm的安装方式从2.1之后就不推荐也不支持了，推荐用docker或者k8s。
编译的话一般只在这几种环境下面编译：https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md#software-requirements

2.0/2.1/2.2这些早期版本的问题挺多，最好别用。

**Reply by zs001122** [reply]
1.因为我部署单机版只是想用于做向量检索，只要求检索的性能。这个也不能用2.0版本的吗？
2.我看了里面提到支持：Debian/Ubuntu、Amazon Linux、MacOS (x86_64)、MacOS (Apple Silicon)，那是不是现在最新的版本对RHEL是不支持的（源码编译，要求不能用docker）？
如果只针对检索性能的需求，又想在RHEL上去源码编译部署，能推荐个版本吗？谢谢！！！

**Reply by zs001122** [reply]
编译文档中https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md#installing-dependencies安装Conan是要求1.64.1，且标注了（目前不支持 Conan 2.x 版本，请使用 1.61 版本），但是我在执行make的时候报了找个问题：
![image](https://github.com/user-attachments/assets/f23389ec-6f83-4245-a158-4cbc84ebb71a)

**Reply by yhmo** [reply]
脚本里指定的是1.64.1，只不过conan本身在安装的时候会给个warning
https://github.com/milvus-io/milvus/blob/a03157838bdc18981d47d3c178683179299fd091/scripts/install_deps.sh#L27

**Reply by zs001122** [reply]
我执行make milvsu后他为什么要下载cmake3.30的，我已经预先升级到3.30了![image](https://github.com/user-attachments/assets/d58cb911-dfc3-4d5a-a17f-554ec13a0e48)

**Reply by yhmo** [reply]
它这个是由conan管理的包，都放在/root/.conan/ 下面，跟你自己安装的那个cmake不是一个东西

**Reply by zs001122** [reply]
我在执行完./scripts/install_deps.py后执行make遇到这个问题：
![image](https://github.com/user-attachments/assets/a7782662-b768-4a0e-94dd-4f83185f9eed)
我的是centos7 ，cmake是3.30，gcc是9.2.0，conan1.64.1，问了gpt说是Google Test没有，但是后面我装了之前还是不行

**Reply by yhmo** [reply]
这个恐怕得自己查了，我们也没遇到过。用报错信息里的一些关键词到google上搜一下看看。

**Reply by zs001122** [reply]
那有centos\REHL相关的单机版源码部署的文档嘛？（是在找不到了。。。）或者2.0.2版本在向量检索上的与最新的版本差距大吗？因为目前的需求只是想利用milvus来在存储跟检索

**Reply by yhmo** [reply]
早期的milvus确实能在centos上编译，后来随着要求的编译环境越来越高，就无法支持了，也没有这方面文档。下面是两条尝试在centos下安装或编译milvus的，基本很难解决。
https://github.com/milvus-io/milvus/discussions/35979
https://github.com/milvus-io/milvus/discussions/30130

2.3/2.4对比2.0在性能上稳定性功能这些方面都是远超的。如果只是通常的存储检索的话，也不必编译吧，还是尽量在docker环境下跑现成的镜像为好。

**Reply by zs001122** [reply]
好的，谢谢！！！
再问一下，就是我这边是只用到存储检索功能，然后数据量是500w条1024维度的向量，单机部署的话服务器配置至少要多少核多少存储？如果想要gpu做加速的话又要什么样的配置？

**Reply by yhmo** [reply]
500万1024维向量size是5x1024x4百万字节=20GB，如果数据是固定的，没有持续性的增删的话，如果使用HNSW/IVF_FLAT索引（占用空间和原始向量的size差不多），用一台32G内存的机器。如果使用IVF_PQ/IVF_SQ8/DISKANN索引（占用空间大约相当于原始向量size的25%-30%），用16G内存的机器。

CPU核数多点就算得快点，核数少点就慢点，4到8核都可以。

要使用GPU版本的话，只能用N卡，compute capability有要求 https://milvus.io/docs/prerequisite-gpu.md。
另外显存(如果用多张卡的话，显存的总和)的大小要能放得下GPU索引，一般GPU_IVF_FLAT占用显存空间是原始向量size的2倍这样。

**Reply by zs001122** [reply]
好的，谢谢！

**Reply by AriesJin** [reply]
哈喽，500万1024维向量size是5x1024x4百万字节=20GB，为什么推荐32G内存的机器的机器呢，是不是因为构建一些标量信息，以及索引本身还需要很多内存，导致实际占用一般是double（看官方的那个推荐）。那24G的能不能吃得消呢

**Reply by xiaofan-luan** [reply]
24G不好说，也可能够，这台机器还要建索引，还要处理实时写入。实在内存很紧张在尝试24GB吧，尤其是你标量很多的情况下，不推荐。

**Comment by xiaofan-luan** [comment]
你可以试试2.4最新版本在你的系统上能否编译过，我们是支持rocky linux8的

**Reply by zs001122** [reply]
rocky linux跟REHL还是有区别的吧，我看2.4的install_deps.h中提到llvm-toolset-11.0-clang，这个用低一点的可以吗？目前我是在centos7上去测试，到时候需要在REHL7上去部署.

**Reply by xiaofan-luan** [reply]
centos7因为已经超过生命周期了所以暂时没有维护。centos8差距应该不大应该是可以维护的。
你可以试试在我们提供的开发镜像上打包，编译出binary之后拿回去跑看看行不行。

**Reply by zs001122** [reply]
再问您个问题，milvus单机版支持BCLinux8.8-kvm-x86_64源码编译吗？它跟centos8、REHL8是兼容的

**Reply by xiaofan-luan** [reply]
centos8应该会比较容易，你可以试试，遇到问题的可以联系小龙我们帮着解决一下
